### ISSUE TITLE
Can we rename InterFAXJerseyClient?

### ISSUE THREAD

**Issue by cbetta** [interfax/interfax-java#9]
@anuragkapur I understand we use Jersey under the hood, but do we really need to make that name appear in the class names?

I'd prefer

``` java
InterFAXClient interFAXClient = new InterFAXJerseyClient("api-username", "api-password");
```

to be 

``` java
InterFAXClient interFAXClient = new InterFAXClient("api-username", "api-password");
```

If possible I'd even prefer it to be more like this:

``` java
InterFAX interfax = new InterFAX("api-username", "api-password");
```

but I might be pushing my ruby-ism here :)

**Comment by anuragkapur** [comment]
@cbetta - I can see the reasoning behind the suggestion. 

To explain a bit, I am using an `Interface` and an `Implementation` here. It is a best practice to programme against interfaces. At the moment there is a single implementation, but I can foresee other implementations using async http client, or a different underlying http library, say apache http client instead of jersey being added in the future. Use of an interface helps in making this happen as the client (of the lib) code can remain the same while one implementation is swapped with another.

So in summary, I have 2 “things" to name here. So how about:

``` java
InterFAX interfax = new DefaultInterFAXClient("api-username", "api-password")
```

better?

**Comment by cbetta** [comment]
Thanks for the explanation, and yes I like that name, and code sample, better.

**Comment by anuragkapur** [comment]
Done https://github.com/interfax/interfax-java/commit/543b8b0b27193eaa23a27c07b937899347723b8c
